envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsParamsInEndpointPathTest.kt

```diff
@@ -0,0 +1,322 @@
+package pl.allegro.tech.servicemesh.envoycontrol.permissions
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.testcontainers.junit.jupiter.Container
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.hasNoRBACDenials
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.hasOneAccessDenialWithActionBlock
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isRbacAccessLog
+import pl.allegro.tech.servicemesh.envoycontrol.config.Echo1EnvoyAuthConfig
+import pl.allegro.tech.servicemesh.envoycontrol.config.Echo2EnvoyAuthConfig
+import pl.allegro.tech.servicemesh.envoycontrol.config.EnvoyControlTestConfiguration
+import pl.allegro.tech.servicemesh.envoycontrol.config.containers.ToxiproxyContainer
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyContainer
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoycontrol.EnvoyControlRunnerTestApp
+
+@SuppressWarnings("LargeClass")
+internal class IncomingPermissionsParamsInEndpointPathTest : EnvoyControlTestConfiguration() {
+
+    companion object {
+        private const val prefix = "envoy-control.envoy.snapshot"
+        private val properties = { sourceClientIp: String ->
+            mapOf(
+                "$prefix.incoming-permissions.enabled" to true,
+                "$prefix.incoming-permissions.source-ip-authentication.ip-from-range.source-ip-client" to
+                    "$sourceClientIp/32",
+                "$prefix.routes.status.create-virtual-cluster" to true
+            )
+        }
+
+
+        // language=yaml
+        private fun proxySettings(unlistedEndpointsPolicy: String) = """
+            node: 
+              metadata: 
+                proxy_settings: 
+                  incoming:
+                    unlistedEndpointsPolicy: $unlistedEndpointsPolicy
+                    endpoints:
+                    - path: "/example-endpoint/{param}/action1"
+                      clients: ["authorized-clients"]
+                    - path: "/example-endpoint/{param}/action1/{param2}/action"
+                      clients: ["authorized-clients"]
+                    - pathPrefix: "/example-endpoint/{param}/action2"
+                      clients: ["authorized-clients"]
+                    - path: "/wildcard-endpoint/{param}/wildcard.*"
```

Comment:
I see we have here two mechanisms:
1. Something a la named params: `{param}`
2. wildcards `*`

Why do we need them both? Wildcard alone is not enough? Is the name of the param in {} used for something?